### PR TITLE
vpd-tool: Correction of VPD Keyword values read from HW

### PR DESCRIPTION
**Pull request by GiridhariKrishnan**
Fixing KW values read from HW in vid-tool.

### CODE REVIEW

**Review by PriyangaRamasamy — COMMENTED** [review]
Please update the commit message with test section

File: ibm_vpd_utils.hpp [hunk 1]
```diff
@@ -350,6 +351,28 @@ inline std::string createBindUnbindDriverCmnd(const std::string& devNameAddr,
             driverType + "/" + bindOrUnbind);
 }
 
+/**
+ * @brief Convert byte array to hex string.
+ * @param[in] vec - byte array
```
Comment (PriyangaRamasamy) [inline ibm_vpd_utils.hpp:356]: its a templated argument. not just byte array. you can change the arg name..

File: ibm_vpd_utils.hpp [hunk 2]
```diff
@@ -350,6 +351,28 @@ inline std::string createBindUnbindDriverCmnd(const std::string& devNameAddr,
...
+ * @return hexadecimal string of bytes.
+ */
+template<typename T>
+std::string byteArrayToHexString(const T& vec)
```
Comment (PriyangaRamasamy) [inline ibm_vpd_utils.hpp:360]: change the api name.

File: ibm_vpd_utils.cpp [hunk 3]
```diff
@@ -371,29 +369,10 @@ vpdType vpdTypeCheck(const Binary& vpdVector)
         // KEYWORD VPD FORMAT
         return vpdType::KEYWORD_VPD;
     }
-    else if (((vpdVector[SPD_BYTE_3] & SPD_BYTE_BIT_0_3_MASK) ==
-              SPD_MODULE_TYPE_DDIMM) &&
-             (is11SFormat.compare(MEMORY_VPD_START_TAG)))
+    else if (is11SFormat.compare(MEMORY_VPD_START_TAG) == 0)
     {
-        // DDIMM Memory VPD format
```
Comment (PriyangaRamasamy) [inline ibm_vpd_utils.cpp:378]: am not following why these ddimm changes are removed as a part of this commit. can you please give a gerrit commit to review with ease

**Review by PriyangaRamasamy — CHANGES_REQUESTED** [review]

File: ibm_vpd_utils.hpp (hunk 1 above)
Comment (GiridhariKrishnan) [reply ibm_vpd_utils.hpp:356]: I'll change the description above and here, parameter should be fine, I think.
```suggestion
 * @param[in] vec - array
```

**Pull request closed without merging**